From 0c1f0e62e72e4c89b0db25aa6f6baa19c7e91e75 Mon Sep 17 00:00:00 2001
From: sirpdboy <herboy2008@gmail.com>
Date: Fri, 13 Sep 2024 19:36:44 +0800
Subject: [PATCH 1/5] luci-mod-system: add modal overlay dialog to reboot

Signed-off-by: sirpdboy <herboy2008@gmail.com>

--- a/modules/luci-mod-system/htdocs/luci-static/resources/view/system/reboot.js
+++ b/modules/luci-mod-system/htdocs/luci-static/resources/view/system/reboot.js
@@ -30,7 +30,27 @@
 		body.appendChild(E('hr'));
 		body.appendChild(E('button', {
 			'class': 'cbi-button cbi-button-action important',
-			'click': ui.createHandlerFn(this, 'handleReboot')
+			'click': function () {
+				ui.showModal(_('Reboot'), [
+					E('p', {}, _('Reboots the operating system of your device')),
+					E('p', {}, _(' ')),
+					E('button', {
+						'class': 'cbi-button cbi-button-action important',
+						'style': 'margin: 2rem 5rem 1rem 5rem; background: red!important; border-color: red!important',
+						'click': function () {
+							ui.hideModal();
+							this.handleReboot();
+						}.bind(this)
+					}, _('OK')),
+					E('button', {
+						'class': 'btn cbi-button cbi-button-apply',
+						'style': 'margin: 0 5rem 1rem 5rem;',
+						'click': function () {
+							ui.hideModal();
+						}
+					}, _('Cancel'))
+				]);
+			}.bind(this)
 		}, _('Perform reboot')));
 
 		return body;
